src/prefect/deployments.py

```diff
@@ -68,11 +69,44 @@ async def run_deployment(
     if scheduled_time is None:
         scheduled_time = pendulum.now("UTC")
 
+    parameters = parameters or {}
+
+    flow_run_ctx = FlowRunContext.get()
+    if flow_run_ctx:
+        # This was called from a flow. Link the flow run as a subflow.
+        from prefect.engine import (
+            Pending,
+            _dynamic_key_for_task_run,
+            collect_task_run_inputs,
+        )
+
+        task_inputs = {
+            k: await collect_task_run_inputs(v) for k, v in parameters.items()
+        }
+
+        # Generate a task in the parent flow run to represent the result of the subflow
+        dummy_task = Task(
+            name=flow_run_ctx.flow.name,
+            fn=flow_run_ctx.flow.fn,
+            version=flow_run_ctx.flow.version,
+        )
+        parent_task_run = await client.create_task_run(
+            task=dummy_task,
+            flow_run_id=flow_run_ctx.flow_run.id,
+            dynamic_key=_dynamic_key_for_task_run(flow_run_ctx, dummy_task),
+            task_inputs=task_inputs,
+            state=Pending(),
+        )
+        parent_task_run_id = parent_task_run.id
```

Comment:
I'd like to consolidate this logic somewhere but would prefer to do so independently of this change.